src/controllers/mainController.ts

```diff
@@ -235,24 +233,46 @@ export default class MainController implements vscode.Disposable {
         if (activeTextEditor && this._vscodeWrapper.isEditingSqlFile) {
             this.onDidOpenTextDocument(activeTextEditor.document);
         }
-        await this.sanitizeProfiles();
+        this.sanitizeConnectionProfiles();
```

Comment:
Why are we not awaiting on this anymore? Won't we potentially run into race condition issues if the sanitization isn't finished but the extension is finished activating and the user tries to modify connections?

Reply:
good catch. https://github.com/microsoft/vscode-mssql/pull/17147